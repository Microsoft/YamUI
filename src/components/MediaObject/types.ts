/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MediaObjectSize {
  XLARGE = 'xLarge',
  LARGE = 'large',
  MEDIUM = 'medium',
  SMALL = 'small',
  XSMALL = 'xSmall',
}
