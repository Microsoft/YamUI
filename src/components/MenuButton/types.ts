/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { ContextualMenuItemType } from 'office-ui-fabric-react/lib/ContextualMenu';

export enum MenuItemType {
  Normal = ContextualMenuItemType.Normal,
  Header = ContextualMenuItemType.Header,
  Divider = ContextualMenuItemType.Divider,
}
