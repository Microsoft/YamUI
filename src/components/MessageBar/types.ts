/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
export enum MessageBarType {
  INFO = 'info',
  WARNING = 'warning',
  ERROR = 'error',
}
