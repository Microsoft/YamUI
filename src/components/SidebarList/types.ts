/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { BaseIllustration } from '../Illustration';

export type SidebarListSize = 'small' | 'medium' | 'large';

export interface BaseSidebarListItem {
  /**
   * A unique key for the item.
   */
  key: string;

  /**
   * The visible title text.
   */
  title: string;

  /**
   * Metadata text displayed under the title, not displayed at small size.
   */
  description?: string;
}
export interface SidebarListIllustrationItem extends BaseSidebarListItem {
  /**
   * A YamUI Illustration component class. You are responsible for importing and providing the correct size.
   */
  illustration: typeof BaseIllustration;
}
export interface SidebarListImageItem extends BaseSidebarListItem {
  /**
   * Custom image JSX. It should fit the list item's pixel width at the given size, or fill 100% of its container width.
   */
  imageContent: React.ReactNode;
}
export type SidebarListItemWithGraphic = SidebarListIllustrationItem | SidebarListImageItem;

export type SidebarListClickableItem = SidebarListItemWithGraphic & {
  /**
   * Callback to be triggered when the item is clicked/invoked.
   */
  onClick(key: BaseSidebarListItem['key']): void;
};
export type SidebarListLinkItem = SidebarListItemWithGraphic & {
  /**
   * The URL this item links to.
   */
  href: string;

  /**
   * If true, opens the link securely in a new window.
   */
  newWindow?: boolean;
};
export type SidebarListItem = SidebarListClickableItem | SidebarListLinkItem;
