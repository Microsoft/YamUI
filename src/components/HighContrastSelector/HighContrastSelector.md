### Notes for use

Add here any notes on how to use the component, or delete this section.

### Examples

Basic usage:

```js { "props": { "data-description": "basic" } }
<HighContrastSelector
  regular="ReactNode"
  highContrast="highContrast ReactNode"
/>
```
