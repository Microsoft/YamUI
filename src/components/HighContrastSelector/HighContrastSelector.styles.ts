/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { mergeStyleSets } from 'office-ui-fabric-react/lib/Styling';
import { memoizeFunction } from 'office-ui-fabric-react/lib/Utilities';

export const getClassNames = memoizeFunction(() => {
  return mergeStyleSets({
    regular: {
      display: 'block',
      selectors: {
        '@media screen and (-ms-high-contrast: active)': {
          display: 'none',
        },
      },
    },
    highContrast: {
      display: 'none',
      selectors: {
        '@media screen and (-ms-high-contrast: active)': {
          display: 'block',
        },
      },
    },
  });
});
