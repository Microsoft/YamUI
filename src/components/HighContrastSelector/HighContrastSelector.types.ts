/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import { BaseComponentProps } from '../../util/BaseComponent/props';

export interface HighContrastSelectorProps extends BaseComponentProps {
  /**
   * Element to be rendered.
   */
  regular: React.ReactNode;

  /**
   * High-Contrast equivalent element to be rendered when in high-contrast mode.
   */
  highContrast: React.ReactNode;
}
