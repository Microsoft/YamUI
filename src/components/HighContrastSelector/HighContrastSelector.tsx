/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import '../../yamui';
import * as React from 'react';
import { join } from '../../util/classNames';
import { HighContrastSelectorProps } from './HighContrastSelector.types';
import { getClassNames } from './HighContrastSelector.styles';

/**
 * Enables given component to render it's equivalent high contrast asset
 * via -ms-high-contrast CSS media feature for MS browsers
 */
export class HighContrastSelector extends React.Component<HighContrastSelectorProps> {
  public render() {
    const { regular, highContrast, className } = this.props;
    const classNames = getClassNames();

    return (
      <div className={join(['y-highContrastSelector', className])}>
        <div className={classNames.regular}>{regular}</div>
        <div className={classNames.highContrast}>{highContrast}</div>
      </div>
    );
  }
}

export default HighContrastSelector;
