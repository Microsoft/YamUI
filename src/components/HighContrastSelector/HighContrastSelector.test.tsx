/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { HighContrastSelector } from './HighContrastSelector';
import { HighContrastSelectorProps } from './HighContrastSelector.types';

describe('<HighContrastSelector />', () => {
  let component: ShallowWrapper<HighContrastSelectorProps>;

  describe('render', () => {
    beforeEach(() => {
      component = shallow(<HighContrastSelector regular={'ReactNode'} highContrast={'High Contrast ReactNode'} />);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
