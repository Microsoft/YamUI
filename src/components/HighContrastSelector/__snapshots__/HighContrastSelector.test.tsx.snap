// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighContrastSelector /> render matches its snapshot 1`] = `
<div
  className="y-highContrastSelector"
>
  <div
    className=

        {
          display: block;
        }
        @media screen and (-ms-high-contrast: active){& {
          display: none;
        }
  >
    ReactNode
  </div>
  <div
    className=

        {
          display: none;
        }
        @media screen and (-ms-high-contrast: active){& {
          display: block;
        }
  >
    High Contrast ReactNode
  </div>
</div>
`;
